runtime/hydradx/src/evm/precompiles/chainlink_adapter.rs

```diff
@@ -197,18 +193,13 @@ pub fn is_oracle_address(address: H160) -> bool {
 	&address.to_fixed_bytes()[0..3] == oracle_address_prefix
 }
 
-/// Converts pallet_ema_oracle::Price to U256. The price is stored as integer, integer part + fractional part.
-/// The fractional part contains `decimals` number of decimal places.
-/// E.g. 123.456789 is stored as 123456 if three decimals are used.
-fn convert_price_to_u256(price: Price, decimals: u8) -> Result<U256, PrecompileFailure> {
-	// avoid panic in exponentiation. Max 256bit number has 78 digits.
-	if decimals > 70 {
-		return Err(PrecompileFailure::Error {
-			exit_status: pallet_evm::ExitError::Other("Too many decimals".into()),
-		});
-	};
+/// Converts pallet_ema_oracle::Price to U256. The price is stored as one integer: integer part + fractional part.
+/// The fractional part contains 8 decimals.
+/// E.g. 7.1234 is stored as 712_340_000.
+fn normalize_price_to_u256(price: Price) -> Result<U256, PrecompileFailure> {
+	let decimals: usize = 8;
 
-	U256::exp10(decimals.into())
+	U256::exp10(decimals)
 		.checked_mul(price.n.into())
 		.ok_or(PrecompileFailure::Error {
 			exit_status: pallet_evm::ExitError::Other("Price conversion failed.".into()),
```

Comment:
this cannot ever happen, you have to scale number and descale them safely

Reply:
see https://github.com/galacticcouncil/hydration-node/blob/174ac5d13f0eca23b120b61ed57b854079d2f939/runtime/adapters/src/lib.rs#L571 for inspiration